legend-depot-store-metrics/src/main/java/org/finos/legend/depot/store/metrics/services/QueryMetricsHandler.java

```diff
@@ -51,22 +60,30 @@ public Optional<VersionQueryMetric> getSummary(String groupId, String artifactId
     }
 
 
-    public List<VersionQueryMetric> getSummaryByProjectVersion()
+    public List<VersionQueryMetric> getSummaryByProjectVersion(boolean sorted)
+    {
+        Stream<VersionQueryMetric> all = metricsStore.getAll().stream()
+                .collect(Collectors.groupingBy(versionQueryMetric -> new ProjectVersion(versionQueryMetric.getGroupId(), versionQueryMetric.getArtifactId(), versionQueryMetric.getVersionId()),
+                         Collectors.collectingAndThen(Collectors.maxBy(MOST_RECENTLY_QUERIED_TIME), Optional::get))).values().stream();
+        return sorted ? all.sorted(MOST_RECENTLY_QUERIED).collect(Collectors.toList()) : all.collect(Collectors.toList());
+    }
+
+    public void consolidateMetrics()
     {
-        Map<ProjectVersion, VersionQueryMetric> all = new HashMap<>();
-        metricsStore.getAll().forEach(m ->
+        List<VersionQueryMetric> versionQueryMetric = getSummaryByProjectVersion(false);
+        LOGGER.info("Started consolidating metrics for all project versions");
+        versionQueryMetric.forEach(metric ->
         {
-            ProjectVersion key = new ProjectVersion(m.getGroupId(), m.getArtifactId(), m.getVersionId());
-            VersionQueryMetric inSummary = all.getOrDefault(key, m);
-            if (inSummary.getLastQueryTime().before(m.getLastQueryTime()))
+            try
             {
-                all.put(key, m);
+                long deletedResult = metricsStore.consolidate(metric);
```

Comment:
metrics are constanly being persisted so yor data is a moving target... and you could be deleting recent daata...you have to consolidate metrcis in the past .. to avoid that